tests/ImageSharp.Tests/Image/PixelAccessorTests.cs

```diff
@@ -14,6 +15,114 @@ namespace ImageSharp.Tests
     /// </summary>
     public class PixelAccessorTests
     {
+        public static Image<TColor> CreateTestImage<TColor>(GenericFactory<TColor> factory)
+            where TColor : struct, IPackedPixel, IEquatable<TColor>
+        {
+            Image<TColor> image = factory.CreateImage(10, 10);
+
+            using (var pixels = image.Lock())
+            {
+                for (int i = 0; i < 10; i++)
+                {
+                    for (int j = 0; j < 10; j++)
+                    {
+                        Vector4 v = new Vector4(i, j, 0, 1);
+                        v /= 10;
+
+                        TColor color = default(TColor);
+                        color.PackFromVector4(v);
+
+                        pixels[i, j] = color;
+                    }
+                }
+            }
+            return image;
+        }
+
+        [Theory]
+        [WithMemberFactory(nameof(CreateTestImage), PixelTypes.All, ComponentOrder.XYZ)]
+        [WithMemberFactory(nameof(CreateTestImage), PixelTypes.All, ComponentOrder.ZYX)]
+        [WithMemberFactory(nameof(CreateTestImage), PixelTypes.All, ComponentOrder.XYZW)]
+        [WithMemberFactory(nameof(CreateTestImage), PixelTypes.All, ComponentOrder.ZYXW)]
+        public void CopyTo_Then_CopyFrom_OnFullImageRect<TColor>(TestImageProvider<TColor> provider, ComponentOrder order)
+            where TColor : struct, IPackedPixel, IEquatable<TColor>
+        {
+            var src = provider.GetImage();
+
+            var dest = new Image<TColor>(src.Width, src.Height);
+
+            using (PixelArea<TColor> area = new PixelArea<TColor>(src.Width, src.Height, order))
+            {
+                using (var srcPixels = src.Lock())
+                {
+                    srcPixels.CopyTo(area, 0, 0);
+                }
+
+                using (var destPixels = dest.Lock())
+                {
+                    destPixels.CopyFrom(area, 0, 0);
+                }
+            }
+
+            Assert.True(src.IsEquivalentTo(dest, false));
+        }
+
+        // TODO: Need a processor in the library with this signature
+        private static void Fill<TColor>(Image<TColor> image, Rectangle region, TColor color)
+             where TColor : struct, IPackedPixel, IEquatable<TColor>
+        {
+            using (var pixels = image.Lock())
+            {
+                for (int y = region.Top; y < region.Bottom; y++)
+                {
+                    for (int x = region.Left; x < region.Right; x++)
+                    {
+                        pixels[x, y] = color;
+                    }
+                }
+            }
+        }
+
+        [Theory]
```

Comment:
Many components of the processors could be tested this way. Unfortunately I'm not yet familiar with their code.